src/lib/quest.ts

```diff
@@ -0,0 +1,72 @@
+import { type Rating } from "./rating";
+
+export interface Quest<TConditionData, TState> {
+  type: QuestType;
+  conditionData: TConditionData;
+  createdAt: Date;
+  playerId: string;
+  state: TState;
+  description: () => string;
+  evaluate: (event: MatchWithPlayers) => QuestStatus;
+  reward: () => Rating;
+  penalty: () => Rating;
+}
+
+export enum QuestType {
+  WinStreak = "WinStreak", // Win X times in a row
+  WinCount = "WinCount", // Win X times
+  WinByPoints = "WinByPoints", // Win by more than X points
+  WinAgainst = "WinAgainst", // Win against specific opponent(s)
+  WinAgainstByPoints = "WinAgainstByPoints", // Win against specific opponent(s) by more than X points
+  WinWith = "WinWith", // Win with specific player(s)
+  PlayMatch = "PlayMatch", // Play a match
+  PlayMatchCount = "PlayMatchCount", // Play X matches
+  PlayMatchWith = "PlayMatchWith", // Play a match with specific player(s)
+}
+
+export enum QuestStatus {
+  InProgress = "InProgress",
+  Completed = "Completed",
+  Failed = "Failed",
+}
+
+export class QuestManager<TCondition, TState> {
```

Comment:
Er idéen at QuestManager altid skal holdes i live in-memory? Eller vil den én gang om ugen løbe igennem alle matches fra den uge?

Reply:
Min tanke var en gang om ugen ville et flow køre.

1. Løb alle matches igennem og se hvor mange quests blev completed
2. Tilføj nye quests
   1. Fejl de ældeste quests hvis spilleren er over quest limit